.github/workflows/ci.yml

```diff
@@ -11,6 +11,20 @@ env:
   RUSTFLAGS: -D warnings
 
 jobs:
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+      - run: cargo clippy --all --all-features
```

Comment:
It doesn't actually deny warnings? This has `--all` which I don't normally use, do you mean `--all-targets`?

Reply:
`--all` is equivalent to `--workspace` (but shorter). It was deprecated in favour of latter because of the confusion potential with `--all-targets` :).

I think `RUSTFLAGS` handles the denying of warnings already, so there is no need to also ask it in the CLI.

Reply:
But `--workspace` is also the default already, so might as well omit it IMO.

Reply:
I've omitted it from the clippy task added in this branch.